Ensure file is a BIDS file before parsing the subject value

constellates opened the pull request:
Potential fix for #250. See more info in issue.

chrisfilo commented:
I would recommend refactoring the subject and session "detectors" to use strict regular expressions looking  only for the appropriately named files (top level for subjects, level below the top for sessions, no extra suffixes or prefixes). It would be more robust.

constellates commented:
That makes sense. I've added updates to use this strategy. I added a `getPathValues` method to the type utils. It will currently return and sub- and ses- values found in the appropriate directory names, but we could expand this to grab other path values. It doesn't check that a path is valid, only that the values are in the correct spots, but the isBIDS check is run before this when generating summary values.

Take a look at the regexs to see if they're reasonable. I'm not super well versed in regex.
